### PR TITLE
rgw: fix opslog cannot record remote_addr

### PR DESCRIPTION
**Pull request by mikulely**
Fixes: http://tracker.ceph.com/issues/20931

Signed-off-by: Jiaying Ren <jiaying.ren@umcloud.com>

### CODE REVIEW

**Comment by mikulely** [comment]
Jenkins, retest this please.

File: src/rgw/rgw_civetweb.cc [hunk 1]
```diff
@@ -120,6 +120,10 @@ void RGWCivetWeb::init_env(CephContext *cct)
     env.set("REMOTE_USER", info->remote_user);
   }
 
+  if (info->remote_addr) {
```
Comment (joscollin) [inline src/rgw/rgw_civetweb.cc]: Would it be better to check the string length too ?

`strlen(info->remote_addr) != 0` 
or 
`if(info->remote_addr[0] != '\0')`

Comment (mikulely) [inline src/rgw/rgw_civetweb.cc]: Even if `remote_add` is empty, it's still fine(ops log entry default value is empty string), I'm fine with adding length check here, the concern is if we've introduced the length check, should we also do this for `info->remote_user`? 

So in order to keep consistency with surrounding code, I'd prefer we don't add length check.

Comment (mattbenjamin) [inline src/rgw/rgw_civetweb.cc]: @mikulely agree

Comment (joscollin) [inline src/rgw/rgw_civetweb.cc]: @mikulely Agreed with your point. But I don't think there is a possibility of `if (info->remote_addr)` would take a false branch. Check `struct mg_request_info{};`.

Comment (mikulely) [inline src/rgw/rgw_civetweb.cc]: @joscollin Agree, dropped.

**Comment by mikulely** [comment]
Jenkins, retest this please.

**Comment by mikulely** [comment]
Jenkins, retest this please.

**Comment by joscollin** [comment]
Jenkins, retest this please.

**Comment by joscollin** [comment]
Jenkins, retest this please.

**Comment by joscollin** [comment]
Jenkins, retest this please.

**Comment by joscollin** [comment]
Test Result:

- http://pulpito.front.sepia.ceph.com/jcollin-2017-08-11_03:03:58-rgw-wip-jcollin-testing_08-10-2017-distro-basic-smithi/
- http://pulpito.front.sepia.ceph.com/compare/?branch=wip-jcollin-testing_08-10-2017&suite=rgw

**Comment by joscollin** [comment]
Jenkins retest this please

**Pull request merged**